raocp/core/scenario_tree.py

```diff
@@ -243,6 +252,7 @@ def __init__(self, transition_prob, initial_distribution, num_stages, stopping_t
         :param num_stages: total number of stages or horizon of the scenario tree
         :param stopping_time: stopping time, which must be no larger than the number of stages [default: None]
         """
+        self.__factory_type = "MarkovChain"
```

Comment:
We shouldn't use strings to store types. The idea of OOP is that all objects have a type.